File: src/main/java/chloe/movietalk/common/HalfPointStepValidator.java
package chloe.movietalk.common;

import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;

public class HalfPointStepValidator implements ConstraintValidator<HalfPointStep, Double> {
    @Override
    public boolean isValid(Double value, ConstraintValidatorContext context) {
        if (value == null) return true;

        return Math.abs(value * 10 % 5) < 0.0001;
    }
}

File: src/main/java/chloe/movietalk/common/HalfPointStepValidator.kt
package chloe.movietalk.common

import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext
import kotlin.math.abs

class HalfPointStepValidator : ConstraintValidator<HalfPointStep?, Double?> {
    override fun isValid(value: Double?, context: ConstraintValidatorContext?): Boolean {
        if (value == null) return true

        return abs(value * 10 % 5) < 0.0001
    }
}
